agent-control/src/k8s/store.rs

```diff
@@ -106,21 +106,28 @@ impl K8sStore {
         #[allow(clippy::readonly_write_lock)]
         let _write_guard = self.rw_lock.write().unwrap();
 
-        let configmap_name = K8sStore::build_cm_name(agent_id, CM_NAME_OPAMP_DATA_PREFIX);
+        let configmap_name = K8sStore::build_cm_name(
+            agent_id,
+            Self::build_cm_prefix(FOLDER_NAME_FLEET_DATA).as_str(),
+        );
         self.k8s_client
             .delete_configmap_key(&configmap_name, self.namespace.as_str(), key)
     }
 
     pub fn build_cm_name(agent_id: &AgentID, prefix: &str) -> String {
         format!("{prefix}{agent_id}")
     }
+
+    pub fn build_cm_prefix(base_name: &str) -> String {
```

Comment:
instead of adding this function and callint it from multiple places we caould change
```
    pub fn build_cm_name(agent_id: &AgentID, prefix: &str) -> String {
        format!("{prefix}{agent_id}")
    }
```

to

```
    pub fn build_cm_name(agent_id: &AgentID, prefix: &str) -> String {
        format!("{prefix}-{agent_id}")
    }
```